packages/test/test-version-utils/src/itSkipsOnFailure.ts

```diff
@@ -0,0 +1,47 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { TestObjectProvider } from "@fluidframework/test-utils";
+// eslint-disable-next-line import/no-extraneous-dependencies
+import { Context } from "mocha";
+import { TestDriverTypes } from "@fluidframework/test-driver-definitions";
+
+function createSkippedTestsWithDriverType(
+	skippedDrivers: TestDriverTypes[],
+	test: Mocha.AsyncFunc,
+) {
+	return async function (this: Context) {
+		const provider: TestObjectProvider | undefined = this.__fluidTestProvider;
+		if (provider === undefined) {
+			throw new Error("Expected __fluidTestProvider on this");
+		}
+		try {
+			await test.bind(this)();
+		} catch (error) {
+			if (skippedDrivers.includes(provider.driver.type)) {
+				provider.logger.sendErrorEvent({ eventName: "TestFailedbutSkipped" }, error);
```

Comment:
@anthony-murphy @Abe27342 I made an event be logged when the skipping occurs, which enables us to detect if the tests on tinylicious/t9s are actually failing. I think it might be satisfactory enough for the current requirements..? (A new task is created to track this [AB#4997](https://dev.azure.com/fluidframework/internal/_workitems/edit/4997))

And IMO we can enhance this function to include additional skip conditions, not just limited to driver types. I propose addressing this as a separate task in the future.

Reply:
sounds reasonable to me.